rcl_action/src/rcl_action/graph.c

```diff
@@ -122,7 +122,10 @@ _filter_action_names(
   // Cleanup if there is an error
   if (RCL_RET_OK != ret) {
     rcl_ret_t fini_ret = rcl_names_and_types_fini(action_names_and_types);
-    (void)fini_ret;  // Error already set
+    if (RCL_RET_OK != fini_ret) {
+      RCUTILS_SAFE_FWRITE_TO_STDERR(
+        "Freeing names and types failed while handling a previous error. Leaking memory!");
```

Comment:
RCUTILS_SAFE_FWRITE_TO_STDERR doesn't append a newline character; we should probably add one here and to the other added messages in this PR.

Reply:
Good call, done in 3fcb531.